sdk/ai/azure-ai-inference/samples/sample_chat_completions_streaming_with_tools.py

```diff
@@ -0,0 +1,202 @@
+            endpoint = os.environ["AZURE_AI_CHAT_ENDPOINT"]
+            key = os.environ["AZURE_AI_CHAT_KEY"]
+
+    except KeyError:
+        print("Missing environment variable 'AZURE_AI_CHAT_ENDPOINT' or 'AZURE_AI_CHAT_KEY'")
+        print("Set them before running this sample.")
+        exit()
+
+    from azure.ai.inference import ChatCompletionsClient
+    from azure.ai.inference.models import (
+        AssistantMessage,
+        ChatCompletionsToolCall,
+        ChatCompletionsToolDefinition,
+        FunctionCall,
+        FunctionDefinition,
+        SystemMessage,
+        ToolMessage,
+        UserMessage,
+    )
+    from azure.core.credentials import AzureKeyCredential
+
+    # Define a function that retrieves flight information
+    def get_flight_info(origin_city: str, destination_city: str):
+        """
+        This is a mock function that returns information about the next
+        flight between two cities.
+
+        Parameters:
+        origin_city (str): The name of the city where the flight originates.
+        destination_city (str): The destination city.
+
+        Returns:
+        str: The airline name, fight number, date and time of the next flight between the cities, in JSON format.
+        """
+        if origin_city == "Seattle" and destination_city == "Miami":
+            return json.dumps({
+                "airline": "Delta",
+                "flight_number": "DL123",
+                "flight_date": "May 7th, 2024",
+                "flight_time": "10:00AM"})
+        return json.dumps({"error": "No flights found between the cities"})
+
+    # Define a function 'tool' that the model can use to retrieves flight information
+    flight_info = ChatCompletionsToolDefinition(
+        function=FunctionDefinition(
+            name="get_flight_info",
+            description="Returns information about the next flight between two cities. This includes the name of the airline, flight number and the date and time of the next flight, in JSON format.",
+            parameters={
+                "type": "object",
+                "properties": {
+                    "origin_city": {
+                        "type": "string",
+                        "description": "The name of the city where the flight originates",
+                    },
+                    "destination_city": {
+                        "type": "string",
+                        "description": "The flight destination city",
+                    },
+                },
+                "required": ["origin_city", "destination_city"],
+            },
+        )
+    )
+
+    if use_azure_openai_endpoint:
+        # Create a chat completion client for Azure OpenAI endpoint
+        client = ChatCompletionsClient(
+            endpoint=endpoint,
+            credential=AzureKeyCredential(""),  # Pass in an empty value
+            headers={"api-key": key},
+            api_version="2024-06-01",  # AOAI api-version. Update as needed.
+        )
```

Comment:
Feels like we should patch this to work with AzureKeyCredential or use Entra ID auth instead

Reply:
See other comment.